### PR TITLE
Remove Mutex in waitservice

### PR DESCRIPTION
**Pull request by majst01**
related to #243

### CODE REVIEW

**Comment by majst01** [comment]
Not so easy:
```
  logger.go:130: 2021-12-07T15:20:05.235Z	ERROR	grpc	finished streaming call with code Internal	{"grpc.start_time": "2021-12-07T15:20:05Z", "system": "grpc", "span.kind": "server", "grpc.service": "v1.Wait", "grpc.method": "Wait", "peer.address": "[::1]:57176", "error": "rpc error: code = Internal desc = unable to cast queue entry to a chan bool", "grpc.code": "Internal", "grpc.time_ms": 7.685}
    common_test.go:53: 
        	Error Trace:	common_test.go:53
        	            				common_test.go:24
        	            				integration_test.go:343
        	            				machine-service_integration_test.go:99
        	Error:      	Received unexpected error:
        	            	json: cannot unmarshal string into Go value of type v1.MachineResponse
        	Test:       	TestMachineAllocationIntegrationFullCycle
```

**Comment by majst01** [comment]
> Not so easy:
> 
> ```
>   logger.go:130: 2021-12-07T15:20:05.235Z	ERROR	grpc	finished streaming call with code Internal	{"grpc.start_time": "2021-12-07T15:20:05Z", "system": "grpc", "span.kind": "server", "grpc.service": "v1.Wait", "grpc.method": "Wait", "peer.address": "[::1]:57176", "error": "rpc error: code = Internal desc = unable to cast queue entry to a chan bool", "grpc.code": "Internal", "grpc.time_ms": 7.685}
>     common_test.go:53: 
>         	Error Trace:	common_test.go:53
>         	            				common_test.go:24
>         	            				integration_test.go:343
>         	            				machine-service_integration_test.go:99
>         	Error:      	Received unexpected error:
>         	            	json: cannot unmarshal string into Go value of type v1.MachineResponse
>         	Test:       	TestMachineAllocationIntegrationFullCycle
> ```

now works

File: cmd/metal-api/internal/grpc/wait-service.go [hunk 1]
```diff
@@ -197,11 +196,15 @@ func sendKeepPatientResponse(srv v1.Wait_WaitServer) error {
...
 func (s *WaitService) handleAllocation(machineID string) {
-	s.queueLock.RLock()
-	defer s.queueLock.RUnlock()
-
-	can, ok := s.queue[machineID]
-	if ok {
+	value, ok := s.queue.Load(machineID)
+	if !ok {
+		return
+	} else {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/grpc/wait-service.go:202]: `else` is not necessary

**Review by Gerrit91 — COMMENTED** [review]
Apart from the one small code-style-nitpicking-comment, it looks much better and also works in the mini-lab.

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**